internal/backend/usageupdater/usagedata.go

```diff
@@ -0,0 +1,105 @@
+package usageupdater
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+	"go.autokitteh.dev/autokitteh/internal/version"
+	"go.autokitteh.dev/autokitteh/internal/xdg"
+	"go.uber.org/zap"
+)
+
+type updateRequest struct {
+	InstallationID string `json:"installationID"`
+	Version        string `json:"version"`
+	Commit         string `json:"commit"`
+	OS             string `json:"os"`
+	Arch           string `json:"arch"`
+}
+
+type UsageUpdater interface {
+	Start()
+	Stop()
+}
+
+type usageUpdater struct {
+	installationID uuid.UUID
+	updateInterval time.Duration
+	shutdownChan   chan struct{}
+	poster         poster
+}
+
+func New(z *zap.Logger, cfg *Config) (UsageUpdater, error) {
+	if !cfg.Enabled {
+		return &nopUpdater{}, nil
+	}
+
+	installationIDFile := filepath.Join(xdg.ConfigHomeDir(), "installID")
+
+	if _, err := os.Stat(installationIDFile); os.IsNotExist(err) {
+		id := uuid.New()
+		if err := os.WriteFile(installationIDFile, []byte(id.String()), 0600); err != nil {
+			return nil, err
+		}
+	}
+
+	data, err := os.ReadFile(installationIDFile)
```

Comment:
maybe if we fail to read and parse we need to recreate file?

Reply:
IDK, 
sounds like a valid idea
I'll add it

Reply:
so this will replace lines 43-48?

Reply:
ack